src/Mvc/test/Mvc.IntegrationTests/ValidationIntegrationTests.cs

```diff
@@ -605,6 +605,78 @@ public async Task Validation_RequiredAttribute_OnPOCOPropertyOfBoundElement_NoDa
         AssertRequiredError("ProductId", error);
     }
 
+#nullable enable
+    private class ParameterInfos
+    {
+        public void Method(
+            string param1,
+            string param2 = "sample_data")
+        {
+        }
+
+        public static ParameterInfo NoAttributesParameterInfo
+            = typeof(ParameterInfos)!
+                .GetMethod(nameof(ParameterInfos.Method))!
+                .GetParameters()[0];
```

Comment:
Nit: Is `NoAttributesParameterInfo` the right name? Isn't this parameter without the default value that has the inferred `[Required]` attribute?

Reply:
It will be inferred or not during the test method.

Reply:
I think we could call it `NonNullableParameterInfo`, is that makes more sense?

Reply:
What attribute does `param2` have?

Reply:
`param2` has no attributes too, however, it has the default value set.

Reply:
`NonNullableParameterInfo` works. Or maybe `RequiredParameterInfo` and `DefaultValueParameterInfo`.

Reply:
Done.